### PR TITLE
RSSFE-13 MUTATIONS

### PR DESCRIPTION
**Pull request by besovadevka**
Added join, leave, createTeam, updateTeam and expel mutations
Close: #37

### CODE REVIEW

File: src/components/ModalExpel/index.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import React, { FC } from 'react';
+import React, { FC, SyntheticEvent } from 'react';
```
Comment (besovadevka) [inline src/components/ModalExpel/index.tsx:1]: remove

File: src/components/ModalExpel/index.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export const ModalExpel: FC<Props> = ({
 }) => {
   const onSubmitModal = () => {
     onClose();
-    if (onSubmit) onSubmit('Modalexpel');
+    if (onSubmit) onSubmit();
```
Comment (manuminsk) [inline src/components/ModalExpel/index.tsx:25]: I'd recommend using curly braces for consistency

File: src/graphql/mutations/addUserToTeamMutation.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      courseIds
+      teamIds
+      teams {
+        id
+        password
+        number
+        courseId
+        socialLink
+        memberIds
+        members {
```
Comment (manuminsk) [inline src/graphql/mutations/addUserToTeamMutation.ts:31]: memberIds, members - are those two fields necessary at the same time?
Also, I'd ask you not to use nested entities if it's not really needed to decrease amount of database calls.

**Review by manuminsk — APPROVED** [review]

File: src/components/ModalExpel/index.tsx (hunk 2 above)
Comment (manuminsk) [reply src/components/ModalExpel/index.tsx:25]: `if (onSubmit) {
    onSubmit();
}`

Comment (besovadevka) [reply src/components/ModalExpel/index.tsx:25]: Done

File: src/graphql/mutations/addUserToTeamMutation.ts (hunk 3 above)
Comment (besovadevka) [reply src/graphql/mutations/addUserToTeamMutation.ts:31]: I'll take this into account

File: src/modules/EditProfile/formFields.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      maxLength: {
+        value: 5,
+        message: 'This input exceed maxLength.',
+      },
+    },
+  },
+];
+
+/*
+<InputField
```
Comment (MadaShindeInai) [inline src/modules/EditProfile/formFields.ts:139]: remove commented code

File: src/modules/EditProfile/index.tsx [hunk 5]
```diff
@@ -29,10 +29,12 @@ export const EditProfile: FC = () => {
   const loginToken = useSelector(selectToken);
   const userData = useSelector(selectUserData);
 
-  const oldCourses: IOldCourses[] = userData.courses.map((course: Course) => ({
-    ...course,
-    isNew: true,
-  }));
+  const oldCourses: IOldCourses[] = (userData.courses as Course[]).map(
```
Comment (MadaShindeInai) [inline src/modules/EditProfile/index.tsx:32]:
```
  const oldCourses = userData.courses.map((course: Course) => ({
      ...course,
      isNew: true,
    });
```
And in types.ts in User, in courses field we should leave only Courses[] and remove | [];

File: src/modules/TeamsList/components/Teams/components/MemberListToggle/styled.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import styled from 'styled-components';
-import { ReactComponent as ChevronArrow } from 'assets/svg/chevron-arrow.svg';
+// import { ReactComponent as ChevronArrow } from 'assets/svg/chevron-arrow.svg';
```
Comment (MadaShindeInai) [inline src/modules/TeamsList/components/Teams/components/MemberListToggle/styled.ts:2]: remove commented code

**Review by MadaShindeInai — CHANGES_REQUESTED** [review]

File: src/graphql/queries/whoAmIQuery.ts [hunk 7]
```diff
@@ -23,8 +23,22 @@ export const WHOAMI_QUERY = gql`
       teamIds
       teams {
         id
+        password
         number
         courseId
+        socialLink
+        memberIds
```
[the remark this answers is not in the document]
Comment (besovadevka) [reply src/graphql/queries/whoAmIQuery.ts:30]: Yes, I am

**Review by MadaShindeInai — APPROVED** [review]

**Review by MadaShindeInai — APPROVED** [review]

**Pull request merged**